docs/DeveloperGuide.md

```diff
@@ -1,34 +1,429 @@
+Step 3. Configure the JDK in IntelliJ Idea to use JDK 11 by following instructions from [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).    
+Step 4. Import the project as a Gradle project.
+Step 5. If you had previously disabled the Gradle plugin, go to `File → Settings → Plugins` to re-enable them.  
+Step 6. Click on Import Project and select the build.gradle file.  
+Step 7. Navigate to the NUSMaze class via the path `src → main → java → seedu.duke → NUSMaze` and right click on it.  
+Step 8. Press run on the `Main()` method of NUSMaze.
 
-## User Stories
+If the set up process had been completed successfully, you should see the following message:  
+![Screenshot 2021-03-25 at 7 03 08 PM](https://user-images.githubusercontent.com/60348727/113017279-e14b9d00-91b1-11eb-8ec3-37c0c3f80475.png)
 
-|Version| As a ... | I want to ... | So that I can ...|
+---------------------------------------------------------------------------------------------
+
+## *2. Design* 
+### 2.1. Architecture
+![img.png](images/architecture.png)
+
+The **Architecture Diagram** above depicts the high-level design of the NUSMaze. You can always refer to this diagram
+to understand how the different components of NUSMaze interact with each other.
+
+The class [`NusMaze`](https://github.com/AY2021S2-CS2113T-T09-2/tp/blob/master/src/main/java/seedu/duke/NusMaze.java) is where the `main()` 
+method belongs and is reponsible for:
+* When the app launches, initialise and connect different components of the NUSMaze in correct sequence.
+* When the app terminates, shut down all the components.
+
+Architecture Components of NUSMaze:
+* [**`UIManager`**](#22-uimanager-component): The user interface of the app
+* [**`Parser`**](#23-parser-component): Processes commands inputted by the user
+* [**`Command`**](#24-command-component): Executes the user command 
+* [**`Router`**](#25-router-component): Searches shortest router
+* [**`Data`**](#26-data-component): Holds the data of the app in memory
+* [**`Storage`**](#27-storage-component): Reads data from and write data to external text files
+
+Explanations on how each component is designed and how it functions are further elaborated in the following 
+chapters of the developer guide.
+
+### 2.2. UIManager Component 
+![img.png](images/ui_design.png)  
+**API**: [UiManager.java](https://github.com/AY2021S2-CS2113T-T09-2/tp/blob/master/src/main/java/seedu/duke/ui/UiManager.java)
+
+The UI of the application is managed by the `UiManager` class as shown by the class diagram above. The individual UI classes for each feature such as `AliasUi`, `DailyRouteUi` and
+`FavouriteUi` extend the `UiManager` class. The UiManager class consists of the methods that are used to display recurrent messages on the *CLI* and also the utilities to get the user's inputs.
+
+The `UiManager` requires the static string variables from the `CommonMessages` class to obtain the commonly used messages that
+such as the divider and input headers.
```

Comment:
"obtain the commonly used messages that such as the divider and input headers." is there a extra "that" before such as ?